### PR TITLE
MUIC-391: [Android] Refactoring queueing flow

### PR DESCRIPTION
**Pull request by deniss-mc**
Messages flow Visitor media state flow TBD
Also usecases for engagement needs fixing

### CODE REVIEW

File: widgetssdk/src/main/java/com/glia/widgets/glia/GliaOnMessageUseCase.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import com.glia.androidsdk.chat.ChatMessage;
 import com.glia.androidsdk.omnicore.OmnicoreEngagement;
+import com.glia.widgets.core.engagement.domain.GliaOnEngagementUseCase;
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/glia/GliaOnMessageUseCase.java:5]: is this needed?

File: widgetssdk/src/main/java/com/glia/widgets/model/MessagesNotSeenHandler.java [hunk 2]
```diff
@@ -2,9 +2,7 @@
 
 import com.glia.androidsdk.chat.Chat;
 import com.glia.androidsdk.chat.ChatMessage;
-import com.glia.androidsdk.omnicore.OmnicoreEngagement;
-import com.glia.widgets.glia.GliaOnEngagementEndUseCase;
-import com.glia.widgets.glia.GliaOnEngagementUseCase;
+import com.glia.widgets.core.engagement.domain.GliaOnEngagementEndUseCase;
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/model/MessagesNotSeenHandler.java:5]: is this needed?

**Review by mustafa-ogun-ozturk — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/model/MessagesNotSeenHandler.java (hunk 2 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/model/MessagesNotSeenHandler.java:5]: yup

File: widgetssdk/src/main/java/com/glia/widgets/glia/GliaOnMessageUseCase.java (hunk 1 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/glia/GliaOnMessageUseCase.java:5]: yup

**Pull request merged**